common/rpc/peer_chooser.go

```diff
@@ -26,21 +26,56 @@ import (
 	"go.uber.org/yarpc/api/peer"
 	"go.uber.org/yarpc/peer/roundrobin"
 
+	"github.com/uber/cadence/common/dynamicconfig"
 	"github.com/uber/cadence/common/log"
+	"github.com/uber/cadence/common/membership"
 )
 
 const defaultDNSRefreshInterval = time.Second * 10
 
 type (
+	PeerChooserOptions struct {
+		// Address is the target dns address. Used by dns peer chooser.
+		Address string
+
+		// ServiceName is the name of service. Used by direct peer chooser.
+		ServiceName string
+
+		// EnableConnectionRetainingDirectChooser is used by direct peer chooser.
+		// If false, yarpc's own default direct peer chooser will be used which doesn retain connections.
+		// If true, cadence's own direct peer chooser will be used which retains connections.
+		EnableConnectionRetainingDirectChooser dynamicconfig.BoolPropertyFn
+	}
 	PeerChooserFactory interface {
```

Comment:
nit, I'd suggest separating out the default DNS implementation into it's own file from the types

Reply:
good idea to split but to avoid the PR growing too much I'd defer.